docs/tutorial/building-a-dataset/index.md

```diff
@@ -0,0 +1,145 @@
+We can now move on to implementing some of the QOF business rules.
+
+In this tutorial, we'll look at rule DM\_006, which is about finding the percentage of patients on the register, with a diagnosis of clinical proteinuria or micro-albuminuria who are currently treated with angiotensin-converting enzymes (ACEs) or angiotensin II receptor blockers (ARBs).
+
+The QOF rules define "currently treated" to mean having a medication within 180 days of the index date.
+
+To do this, we'll use these codelists: [`PRT_COD`][1], [`MAL_COD`][2], [`ACE_COD`][3], and [`ARB_COD`][4].
+
+We have already seen how to determine whether a patient has a clinical event matching a code in a codelist.
+Medications are recorded in a similar way to clinical events, and can be queried using the same kinds of ehrQL.
+
+
+```ehrql
+from ehrql import codelist_from_csv, days
+from ehrql.tables.core import clinical_events, medications
+
+
+index_date = "2024-03-31"
+
+proteinuria_codes = codelist_from_csv("codelists/prt_cod.csv", column="code")
+microalbuminuria_codes = codelist_from_csv("codelists/mal_cod.csv", column="code")
+ace_codes = codelist_from_csv("codelists/ace_cod.csv", column="code")
+arb_codes = codelist_from_csv("codelists/arb_cod.csv", column="code")
+
+has_proteinuria_diagnosis = (
+    clinical_events.where(clinical_events.date.is_before(index_date))
+    .where(clinical_events.snomedct_code.is_in(proteinuria_codes))
+    .exists_for_patient()
+)
+
+has_microalbuminuria_diagnosis = (
+    clinical_events.where(clinical_events.date.is_before(index_date))
+    .where(clinical_events.snomedct_code.is_in(microalbuminuria_codes))
+    .exists_for_patient()
+)
+
+has_arb_treatment = (
+    medications.where(medications.date.is_between(index_date - days(180), index_date))
+    .where(medications.dmd_code.is_in(arb_codes))
+    .exists_for_patient()
+)
+
+has_ace_treatment = (
+    medications.where(medications.date.is_between(index_date - days(180), index_date))
+    .where(medications.dmd_code.is_in(ace_codes))
+    .exists_for_patient()
+)
+
+debug(
+    has_proteinuria_diagnosis,
+    has_microalbuminuria_diagnosis,
+    has_arb_treatment,
+    has_ace_treatment,
+)
+```
+
+So far, we've been using `debug()` to show us what our queries return when running against dummy data.
+
+But we can't use `debug()` to extract data for a real study!
+
+Instead, we need to define a dataset.  As a reminder: a dataset is a new table containing one row per patient, and a dataset definition consists of a population definition and a set of column definitions.
+
+We then use `datatset = create_dataset()` to create a new dataset object, `dataset.set_population(...)` to define the population, and `dataset.column = ...` to define the columns:
```

Comment:
```suggestion
We then use `dataset = create_dataset()` to create a new dataset object, `dataset.set_population(...)` to define the population, and `dataset.column = ...` to define the columns:
```

Reply:
I don't much care for `dataset.column = ...` as a literal reading suggests that they should go ahead and write e.g. this in their ehrQL:

```python
dataset.column = on_register, has_proteinuria_diagnosis
```